Gifski/MainWindowController.swift

```diff
@@ -169,7 +167,7 @@ final class MainWindowController: NSWindowController {
 				case .cancelled:
 					break
 				default:
-					fatalError(error.localizedDescription)
+					self.presentError(SoftError(error), modalFor: self.window)
```

Comment:
This only captures the context here and not where the errors are actually initiated. I considered creating an utility function that would accept the error, save the context to `ErrorContext.store` and then return the error. And then we could get the context from the `ErrorContext.store` later on.

So instead of:

```swift
completionHandlerOnce?(.generateFrameFailed(error))
```

It would be:

```swift
completionHandlerOnce?(ErrorContext.save(.generateFrameFailed(error)))
```

The benefit with this is that `.save()` could also accept a dictionary of custom metadata for even more context.